src/backup.sh

```diff
@@ -9,140 +9,158 @@ function info {
   echo -e "\n$bold[INFO] $1$reset\n"
 }
 
-info "Backup starting"
-TIME_START="$(date +%s.%N)"
-DOCKER_SOCK="/var/run/docker.sock"
-
-if [ ! -z "$BACKUP_CUSTOM_LABEL" ]; then
-  CUSTOM_LABEL="--filter label=$BACKUP_CUSTOM_LABEL"
-fi
-
-if [ -S "$DOCKER_SOCK" ]; then
+if [ "$CHECK_HOST" != "false" ]; then
   TEMPFILE="$(mktemp)"
-  docker ps --format "{{.ID}}" --filter "label=docker-volume-backup.stop-during-backup=true" $CUSTOM_LABEL > "$TEMPFILE"
-  CONTAINERS_TO_STOP="$(cat $TEMPFILE | tr '\n' ' ')"
-  CONTAINERS_TO_STOP_TOTAL="$(cat $TEMPFILE | wc -l)"
-  CONTAINERS_TOTAL="$(docker ps --format "{{.ID}}" | wc -l)"
-  rm "$TEMPFILE"
-  echo "$CONTAINERS_TOTAL containers running on host in total"
-  echo "$CONTAINERS_TO_STOP_TOTAL containers marked to be stopped during backup"
+  ping -c 1 $CHECK_HOST | grep '1 packets transmitted, 1 received' > "$TEMPFILE"
+  PING_RESULT="$(cat $TEMPFILE)"
+  if [ ! -z "$PING_RESULT" ]; then
+    skip="false"
+    echo "$CHECK_HOST is available."
+  else
+    skip="true"
+    echo "$CHECK_HOST is not available."
+    info "Backup skipped"
+  fi
 else
-  CONTAINERS_TO_STOP_TOTAL="0"
-  CONTAINERS_TOTAL="0"
-  echo "Cannot access \"$DOCKER_SOCK\", won't look for containers to stop"
+  skip="false"
 fi
 
-if [ "$CONTAINERS_TO_STOP_TOTAL" != "0" ]; then
-  info "Stopping containers"
-  docker stop $CONTAINERS_TO_STOP
-fi
+if [ "$skip" == "false" ]; then
```

Comment:
Would it be tidier to simply `exit 0` after `info "Backup skipped"`?

That way the rest of the script wouldn't have to be wrapped in a huge `if` block.

What do you think?

Reply:
i also do not like that `if` block. Thank you for point this out. I was not aware that terminating the script is that easy. I will consider that in an upcoming commit.